src/renderer/pages/Project.tsx

```diff
@@ -1,28 +1,80 @@
-import { Stack } from '@mui/material';
-import { Box } from '@mui/system';
+import { Box, Stack } from '@mui/material';
+import { useCallback, useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import StandardButton from 'renderer/components/StandardButton';
-import { projectOpened } from 'renderer/store/actions';
+import TranscriptionBlock from 'renderer/components/TranscriptionBlock';
+import { transcriptionCreated } from 'renderer/store/actions';
+import { Transcription } from 'sharedTypes';
 import { ApplicationStore } from '../store/helpers';
 
+/*
+This is the page that gets displayed while you are editing a video.
+It will be primarily composed of the transcription area, an editable text box whose
+changes get reflected in the video. In addition to that, there is a video preview
+section to the side among other things.
+*/
 const ProjectPage = () => {
+  const dispatch = useDispatch();
   const currentProject = useSelector(
     (store: ApplicationStore) => store.currentProject
   );
 
-  const dispatch = useDispatch();
+  // RK: I really shouldn't use transcriptionCreated here - but i'm lazy and it works
+  const saveTranscription: (transcription: Transcription) => void = useCallback(
+    (transcription) => dispatch(transcriptionCreated(transcription)),
+    [dispatch]
+  );
+
+  const deleteWord = (firstWordIndex: number, numberOfWords: number) => {
+    if (currentProject && currentProject.transcription) {
+      // eslint-disable-next-line no-plusplus
+      for (let i = firstWordIndex; i < firstWordIndex + numberOfWords; i++) {
```

Comment:
Not that you should be doing this anyway, but for reference it can be done as a one liner: `currentProject.transcription.words.slice(firstWordIndex, numberOfWords).forEach(word => word.deleted = true);`